src/backend/pipeline/modules/runnable.py

```diff
@@ -33,3 +33,13 @@ def prepare(self, data: Data):
         Args:
             data: the pipeline data object
         """
+
+    @abstractmethod
+    def upload(self, bucket_name: str, data: Data):
+        """
+        Upload data to Google Cloud Storage.
+
+        Args:
+            bucket_name: unique bucket name from Storage
+            data: the pipeline data object
+        """
```

Comment:
See the comment below about changing `upload` to run from `pipeline::run()`. That'll make it easy to set the bucket path correctly (according to notes from **[Sprint 3 meeting](https://app.clickup.com/9004004983/v/dc/8cawdkq-1447/8cawdkq-1367)**)